src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/provisioner/src/support.py

```diff
@@ -0,0 +1,58 @@
+"""Support module used throughout the ADF
+"""
+
+from botocore.config import Config
+from botocore.exceptions import ClientError, BotoCoreError
+from logger import configure_logger
+
+from .account import Account
+
+
+LOGGER = configure_logger(__name__)
+
+class Support: # pylint: disable=R0904
+    """Class used for accessing AWS Support API
+    """
+
+    _config = Config(retries=dict(max_attempts=30))
+
+    def __init__(self, role):
+        self.client = role.client("support", region_name='us-east-1', config=Support._config)
+
+    def check_aws_support_subscription(self):
+        try:
+            self.client.describe_cases()
+            return True
+        except (ClientError, BotoCoreError) as e:
+            if e.response["Error"]["Code"] == "SubscriptionRequiredException":
+                LOGGER.warning("Failed subscription for account")
+                LOGGER.info('Enterprise Support is not enabled')
+                return False
+            raise e
```

Comment:
Could you change this to become: `raise`?

There is a big difference between `raise e` and `raise`, where `raise e` generated a new stack trace instead of reusing the previous one. See [this StackOverflow answer](https://stackoverflow.com/a/45986945).